classes/Search.php

```diff
@@ -0,0 +1,572 @@
+
+        $sort_order = 'newest';
+        if ( $o == 'o' ) {
+            $sort_order = 'oldest';
+        } else if ( $o == 'r' ) {
+            $sort_order = 'relevance';
+        }
+
+        $members = null;
+        $cons = null;
+        $glossary = null;
+        if ($pagenum == 1 && $args['s'] && !preg_match('#[a-z]+:[a-z0-9]+#', $args['s'])) {
+            $members = $this->find_members($args['s']);
+            $cons = $this->find_constituency($args);
+            $glossary = $this->find_glossary_items($args);
+        }
+
+        if (!defined('FRONT_END_SEARCH') || !FRONT_END_SEARCH) {
+            return array(
+                'error' =>'Apologies, search has been turned off currently for performance reasons.'
+            );
+        }
+
+        if (!$SEARCHENGINE->valid) {
+            return array('error' => $SEARCHENGINE->error);
+        } else {
+            $LIST = new \HANSARDLIST();
+            $data = $LIST->display('search', $args , 'none');
+            $data['sort_order'] = $sort_order;
+            $data['members'] = $members;
+            $data['cons'] = $cons;
+            $data['glossary'] = $glossary;
+            return $data;
+        }
+    }
+
+    private function search_order_p($searchstring) {
+        global $DATA, $this_page;
+
+        $q_house = '';
+        if (ctype_digit(get_http_var('house'))) {
+            $q_house = get_http_var('house');
+        }
+
+        $wtt = get_http_var('wtt');
+        if ($wtt) {
+            $q_house = 2;
+        }
+
+        # Fetch the results
+        $data = search_by_usage($searchstring, $q_house);
+
+        if ($wtt) {
+            $q_house = 2;
+            $pagetitle = 'League table of Lords who say ' . $data['pagetitle'];
+        } else {
+            $pagetitle = 'Who says ' . $data['pagetitle'] . ' the most?';
+        }
+        $DATA->set_page_metadata($this_page, 'title', $pagetitle);
+
+
+        $data['house'] = $q_house;
+        $data['wtt'] = $wtt;
+        return $data;
+    }
+
+    private function find_constituency($args) {
+        if ($args['s'] != '') {
+            $searchterm = $args['s'];
+        } else {
+            return false;
+        }
+
+        list ($constituencies, ) = search_constituencies_by_query($searchterm);
+
+        $constituency = "";
+        if (count($constituencies)==1) {
+            $constituency = $constituencies[0];
+        }
+
+        $cons = array();
+        try {
+            if ($constituency != '') {
+                // Got a match, display....
+
+                $MEMBER = new Member(array('constituency'=>$constituency, 'house' => 1));
+                $cons[] = $MEMBER;
+            } elseif (count($constituencies)) {
+                foreach ($constituencies as $constituency) {
+                    $MEMBER = new Member(array('constituency'=>$constituency, 'house' => 1));
+                    $cons[] = $MEMBER;
+                }
+            }
+        } catch ( MemberException $e ) {
```

Comment:
Should this continue if there's an exception, to get other matches that might work?